app/models/dialog/orchestration_template_service_dialog.rb

```diff
@@ -0,0 +1,139 @@
+class Dialog
+  class OrchestrationTemplateServiceDialog
+    def self.create_dialog(label, template)
+      Dialog::OrchestrationTemplateServiceDialog.new.create_dialog(label, template)
+    end
+
+    def create_dialog(label, template)
+      Dialog.new(:label => label, :buttons => "submit,cancel").tap do |dialog|
+        tab = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
+        add_stack_group(template.deployment_options, tab, 0)
+
+        template.parameter_groups.each_with_index do |parameter_group, index|
+          add_parameter_group(parameter_group, tab, index + 1)
+        end
+        dialog.save!
+      end
+    end
+
+    private
+
+    def add_stack_group(deploy_options, tab, position)
+      tab.dialog_groups.build(
+        :display  => "edit",
+        :label    => "Options",
+        :position => position
+      ).tap do |dialog_group|
+        deploy_options.each_with_index { |opt, index| add_field(opt, dialog_group, index) }
+      end
+    end
+
+    def add_parameter_group(parameter_group, tab, position)
+      return if parameter_group.parameters.blank?
+
+      tab.dialog_groups.build(
+        :display  => "edit",
+        :label    => parameter_group.label || "Parameter Group#{position}",
+        :position => position
+      ).tap do |dialog_group|
+        parameter_group.parameters.each_with_index { |param, index| add_field(param, dialog_group, index, 'param_') }
+      end
+    end
+
+    def add_field(parameter, group, position, prefix = nil)
```

Comment:
Suggest making the `prefix` variable name more descriptive.  Maybe `name_prefix`.